### PR TITLE
Implement preadv and pwritev

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Implement preadv and pwritev
## Testing
build tested with modified wamr

### CODE REVIEW

File: libs/libc/uio/lib_pwritev.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                {
+                  if (pos != (off_t)-1)
+                    {
+                      /* Save the errno value */
+
+                      int save = get_errno();
+
+                      /* Restore the file position */
+
+                      lseek(fildes, pos, SEEK_SET);
```
Comment (xiaoxiang781216) [inline libs/libc/uio/lib_pwritev.c:88]: why need restore file pointer here? pwrite at line 74 doesn't expect to change the file location.

Comment (yamt) [reply libs/libc/uio/lib_pwritev.c]: because of copy-and-paste w/o thinking. :-)
fixed.

**Review by xiaoxiang781216 — APPROVED** [review]

File: libs/libc/uio/lib_pwritev.c (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libc/uio/lib_pwritev.c]: it's better to fix pwrite too, could you create a new patch/pr for this?

Comment (yamt) [reply libs/libc/uio/lib_pwritev.c]: > it's better to fix pwrite too, could you create a new patch/pr for this?

what needs to be fixed in pwrite?

**Pull request merged**